packages/flutterfire_cli/lib/src/commands/upload_symbols.dart

```diff
@@ -0,0 +1,320 @@
+        return file.path;
+      } else {
+        // Update app ID to match the current one from firebase.json
+        map[appIdPropertyName] = appId;
+        final updatedMapJson = json.encode(map);
+        file.writeAsStringSync(updatedMapJson);
+
+        return file.path;
+      }
+    } else {
+      // Create file if it does not exist
+      await file.create(recursive: true);
+      final map = {appIdPropertyName: appId, projectIdPropertyName: projectId};
+
+      final mapJson = json.encode(map);
+
+      file.writeAsStringSync(mapJson);
+
+      return file.path;
+    }
+  }
+
+  Future<ConfigurationResults> _getConfigurationFromFirebaseJsonFile() async {
+    final platformConfig =
+        await appleConfigFromFirebaseJson(appleProjectPath, platform);
+
+    String? appId;
+    String? projectId;
+    bool? uploadDebugSymbols;
+    Map configurationMaps;
+    String keyToConfig;
+    try {
+      configurationMaps = platformConfig[configuration] as Map;
+      Map configurationMap;
+
+      switch (projectConfiguration) {
+        case ProjectConfiguration.buildConfiguration:
+          keyToConfig = buildConfiguration!;
+          // ignore: cast_nullable_to_non_nullable
+          configurationMap =
+              configurationMaps[keyToConfig] as Map<String, dynamic>;
+          break;
+        case ProjectConfiguration.target:
+          keyToConfig = target!;
+          // ignore: cast_nullable_to_non_nullable
+          configurationMap =
+              configurationMaps[keyToConfig] as Map<String, dynamic>;
+          break;
+        case ProjectConfiguration.defaultConfig:
+          keyToConfig = defaultConfig!;
+          // There is no more nested maps in "default" configuration, it is one single default configuration map
+          configurationMap = configurationMaps;
+      }
+
+      uploadDebugSymbols = configurationMap[kUploadDebugSymbols] as bool?;
+      appId = configurationMap[kAppId] as String?;
+      projectId = configurationMap[kProjectId] as String?;
+    } catch (e) {
+      throw FirebaseJsonException();
+    }
+
+    if (projectId == null || appId == null) {
+      throw FirebaseJsonException();
+    }
+
+    return ConfigurationResults(
+      appId,
+      projectId,
+      uploadDebugSymbols,
+      keyToConfig,
+    );
+  }
+
+  @override
+  Future<void> run() async {
+    if (!Platform.isMacOS) return;
+
+    final configurationResults = await _getConfigurationFromFirebaseJsonFile();
+    final uploadDebugSymbols = configurationResults.runDebugSymbolsScript;
+    final appId = configurationResults.appId;
+    final projectId = configurationResults.projectId;
+    final configurationKey = configurationResults.keyToConfig;
+
+    // Exit if the user chooses not to run debug upload symbol script
+    if (uploadDebugSymbols == false || uploadDebugSymbols == null) return;
+    final appIdFileDirectory = path.join(
+      path.dirname(Directory.current.path),
+      '.dart_tool/flutterfire/platforms',
```

Comment:
This needs to be path joined right?

Reply:
Will change but it doesn't make a difference 👍